hazelcast-client-new/src/main/java/com/hazelcast/client/spi/impl/ClientInvocationServiceSupport.java

```diff
@@ -375,24 +375,11 @@ private void handleClientMessage(ClientMessage clientMessage) throws ClassNotFou
             }
 
             if (ExceptionResultParameters.TYPE == clientMessage.getMessageType()) {
-                ExceptionResultParameters exceptionResultParameters = ExceptionResultParameters.decode(clientMessage);
-                Throwable exception;
-                if (exceptionResultParameters.causeClassName != null) {
-                    Class<?> causeClazz = Class.forName(exceptionResultParameters.causeClassName);
-                    Constructor<?> causeConstructor = causeClazz.getDeclaredConstructor(new Class[]{String.class});
-                    causeConstructor.setAccessible(true);
-                    Throwable cause = (Throwable) causeConstructor.newInstance(exceptionResultParameters.message);
-
-                    Class<?> clazz = Class.forName(exceptionResultParameters.className);
-                    Constructor<?> constructor = clazz.getDeclaredConstructor(new Class[]{String.class, Throwable.class});
-                    constructor.setAccessible(true);
-                    exception = (Throwable) constructor.newInstance(exceptionResultParameters.message, cause);
-                } else {
-                    Class<?> clazz = Class.forName(exceptionResultParameters.className);
-                    Constructor<?> constructor = clazz.getDeclaredConstructor(new Class[]{String.class});
-                    constructor.setAccessible(true);
-                    exception = (Throwable) constructor.newInstance(exceptionResultParameters.message);
-                }
+                ExceptionResultParameters exParameters = ExceptionResultParameters.decode(clientMessage);
```

Comment:
We changed all the names from *parameters to *Codec, hence may be we can change ExceptionResultParameters to ErrorCodec to be more consistent.